docs/pages/documentation/3-querying-with-graql/ddl.md

```diff
@@ -0,0 +1,115 @@
+sidebar: documentation_sidebar
+permalink: /documentation/graql/ddl.html
+folder: graql
+---
+
+# Define Query
+
+`define` [`<variable patterns>`](./query.html#variable-pattern)
+
+The [define query](#define-query) will add the given [variable patterns](./query.html#variable-pattern) to the schema.
+
+After execution, it will return a single [answer](./query.html#answer) containing bindings for all
+[variables](./query.html#variable) mentioned in the [variable patterns](./query.html#variable-pattern).
+
+# Undefine Query
+
+`undefine` [`<variable patterns>`](./query.html#variable-pattern)
+
+The [undefine query](#undefine-query) will remove the given [variable patterns](./query.html#variable-pattern) from the
+schema.
+
+In order to remove a schema concept entirely, it is sufficient to undefine its [direct super-concept](#sub):
+
+<!-- Ignored so we don't delete the person type for real -->
+```graql-test-ignore
+undefine person sub entity;
+```
+
+# Supported Properties
+
+Within the [variable patterns](./query.html#variable-pattern) of both [define](#define-query) and
+[undefine](#undefine-query) queries, the following [properties](./query.html#property) are supported:
+
+## id
+
+`$x id <identifier>` will assign `$x` to an existing concept with the given ID. It is an error if no such concept
+exists.
+
+## label
+
+`$A label <identifier>` will assign `$A` to a schema concept with the given label. If no such schema concept exists,
+one will be created.
+
+## sub
+
+`$A sub $B` defines `$A` as the direct sub-concept of `$B`.
+
+## relates
+
+`$A relates $B` define the relationship type `$A` to directly relate the role `$B`.
+
+## plays
+
+`$A plays $B` defines the type `$A` to directly play the role `$B`.
+
+## has (type)
+
+`$A has <identifier>` defines the type `$A` to have attributes of type `<identifier>`.
+
+This is done using the following relationship structure:
+```graql-test-ignore
+has-<identifier>-owner sub has-<sup>-owner;
+has-<identifier>-value sub has-<sup>-value;
+has-<identifier> sub has-<sup>, relates has-<identifier>-owner, relates has-<identifier>-value;
+
+$A plays has-<identifier>-owner;
+<identifier> plays has-<identifier>-value;
+```
+Where `<sup>` is the direct super-concept of `<identifier>`.
+
+## key (type)
+
+`$A key <identifier>` defines the type `$A` to have a key of attribute type `<identifier>`.
+
+This is done using the following relationship structure:
+```graql-test-ignore
+key-<identifier>-owner sub key-<sup>-owner;
+key-<identifier>-value sub key-<sup>-value;
+key-<identifier> sub key-<sup>, relates key-<identifier>-owner, relates key-<identifier>-value;
+
+$A plays<<required>> has-<identifier>-owner;
+<identifier> plays has-<identifier>-value;
+```
+Where `<sup>` is the direct super-concept of `<identifier>`.
+<!-- TODO: This is pretty bad -->
+(note that `plays<<required>>` is not valid syntax, but indicates that instances of the type _must_ play the required
+role exactly once).
+
+## datatype
```

Comment:
I feel that fore these properties it should be better highlighted that they can only be used with certain schema concepts. For example:

```
$A is-abstract` defines the type `$A` to be abstract.
```

Is very subtly telling you only types can be abstract maybe too subtly.

Reply:
Maybe just putting it in bold or italics?

> `$A is-abstract` defines the _type_ `$A` to be abstract.